Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -399,7 +399,13 @@ public void ClearAllDirtyBits()
             syncVarDirtyBits = 0L;
 
             // flush all unsynchronized changes in syncobjects
-            m_SyncObjects.ForEach(obj => obj.Flush());
+            // note: don't use List.ForEach here, this is a hot path
+            // List.ForEach: 432b/frame
+            // for: 231b/frame
+            for (int i = 0; i < m_SyncObjects.Count; ++i)
```

Comment:
can you use foreach(...) please?

Reply:
The problem is that foreach generates more garbage.

Reply:
@vis2k Vincenz099 is right,  foreach generates garbage on Lists.

Reply:
wasn't there a discussion yesterday about foreach (var in list) not making a difference?

Reply:
It behaves very different for different data structures.    foreach is fine for arrays and apparently dictionaries.    I have not profiled it for Lists,  but traditionally,  they were very bad for lists.

Reply:
@rodolphito could you run one more profiling with foreach (var in list) please?